File: 14. Longest Common Prefix/solution.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        count = len(strs)
        if (1 == count):
            return strs[0]

        prefix = ''
        for i in range(len(strs[0])):
            for j in range(1, count):
                if i >= len(strs[j]) or (strs[0][i] != strs[j][i]):
                    return prefix
            prefix = prefix + strs[0][i]

        return prefix

File: 14. Longest Common Prefix/solution.php
<?php
class Solution {

    /**
     * @param String[] $strs
     * @return String
     */
    function longestCommonPrefix($strs) {
        $prefix = '';
        $first = $strs[0];
        $len = strlen($first);
        for ($i=0;$i < $len;$i++) {
            for ($j=1;$j < count($strs);$j++) {
                if ($first[$i] != $strs[$j][$i]) {
                    return $prefix;
                }
            }
            $prefix .= $first[$i];
        }

        return $prefix;
    }
}
